### ISSUE TITLE
Show the HDD speed rate on the 'out of sync items' modal

### ISSUE THREAD

**Issue by bigbigbang** [syncthing/syncthing#3985]
As syncthing re-use existing data instead of downloading them when the data block is already on disk, I think it will be great to see the 'copying' speed rate too.

In my case i'm syncing windows install images in several languages, and syncthing detects a lot of 'already exsting data block' in these files. When I monitor the average speed of the sync, the download speed rate isn't really relevant.
There might be different ways to show that in GUI.. i think we must not do a global speed average (mixing downloading and writing rates) as it would say nothing (and probably getting inconsistent results).

**Comment by calmh** [comment]
I doubt we will that fine grained. But I could imagine should disk read/write rate per folder in the folder listing and as a sum for the entire device in the "local device" section - mirroring what we do with network rates for devices.

**Comment by bigbigbang** [comment]
Yes ^ it would be already better than nothing :)
The current writing speed for a device is enough for monitoring (and to see what syncthing is doing when not downloading).

**Comment by AudriusButkevicius** [comment]
Sorry, but this is pointless. if you want to know how fast is your disk, run benchmarks.

**Comment by Ferroin** [comment]
No, the point is knowing how much of your disk bandwidth Syncthing is using, which is a perfectly reasonable thing to care about.

**Comment by bigbigbang** [comment]
Yes, could be like the CPU rate ^
In fact, right now, when syncthing is copying data (from elsewhere) to disk, we don't see it clearly on the GUI (my opinion). It could be interesting to see the % of HDD usage (and maybe speed rate too). 
Like this, when download speed are low, we can correlate this with a high HDD usage.

**Comment by calmh** [comment]
Yeah I think this is less about benchmarking and more about showing that something is happening, especially during large copy phases.

**Comment by AudriusButkevicius** [comment]
You know that something is happening as the progress bar moves?

**Comment by Ferroin** [comment]
The progress bar doesn't always move meaningfully while something is happening, especially if you have a very large number of files.  Even if the progress bar is moving, it's still useful at times to know exactly _what_ is happening.

Additionally, Syncthing already reports every other major resource usage metric in the UI (CPU, RAM, Network utilization), so it seems kind of illogical that it doesn't report disk usage (especially considering that it's trivial info to get programmatically on Linux and shouldn't be all that hard on Android, Solaris, and OS X).

**Comment by calmh** [comment]
I think we should get it by counting our reads and writes instead of asking the OS. It also gets us a metric we can use to calculate estimated completion time, as our average write speed for a folder is directly comparable to the amount of data left to sync.

**Comment by Ferroin** [comment]
Counting I/O calls is probably the best cross-platform way of doing this, but I'd advise against exposing IOPS instead of actual bandwidth usage to users as the disk utilization metric.  IOPS is a very abstract measurement for end-users.  It makes a lot of sense in some contexts (database systems for example), but near zero in many others (file transfers for example), and I'd argue that Syncthing's usage is a case where it really doesn't make much sense as a resource metric.  There is of course a bit more accounting involved to compute actual bandwidth usage, but the math isn't hard (you'd be doing most of it anyway when calculating IOPS), and I doubt that the accounting overhead will be all that great.

**Comment by calmh** [comment]
Yeah, by counting reads and writes I meant bytes read and written. IOPS indeed doesn't make sense as a metric on the application level, really.

**Comment by AudriusButkevicius** [comment]
Do we still want to do this? Given we removed CPU usage and what not.

**Comment by imsodin** [comment]
This falls more into the "transfer rate" category, which we do control and manage. Thus I think it's in scope. Still very much optional and completely not of interest to me, but in my opinion a sensible PR should be acceptable.
